examples/allocation-endpoint/client/main.go

```diff
@@ -0,0 +1,152 @@
+// Package main runs an Allocation Endpoint client with configurations for benchmarking
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	pb "agones.dev/agones/pkg/allocation/go"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	port = "443"
+)
+
+var (
+	clientCount    = flag.Int("clientCount", 10, "Number of clients to perform allocations")
+	perClientCalls = flag.Int("perClientCalls", 200, "Per Client allocations running in parallel")
+	runs           = flag.Int("runs", 10, "The number of times the experiment is run and averaged across")
+	serverAddr     = flag.String("url", "", "The Cloud Run URL for allocation endpoint.")
+	logger         = log.New(os.Stdout, "", 0)
+	muErr          sync.Mutex // gaurds error variable update
+	muSucc         sync.Mutex // guards success variable update
+)
+
+//go:embed sa_key.json
+var saKey string
+
+func main() {
+	flag.Parse()
+
+	logger.Printf("Running %d times averaged across %d allocation calls per %d clients\n", *runs, *perClientCalls, *clientCount)
+
+	var errorCalls, successCalls int64
+	var successLatency float64
+	errorGroup := make(map[string]int64)
+	startRun := time.Now()
+	waitBetweenRunsInSec := time.Second * 3
+
+	audience := fmt.Sprintf("https://%s", *serverAddr)
+	ts, err := idtoken.NewTokenSource(context.Background(), audience, idtoken.WithCredentialsJSON([]byte(saKey)))
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	for k := 0; k < *clientCount; k++ {
+		wg.Add(1)
+		go func(k int) {
+			defer wg.Done()
+			cred := credentials.NewTLS(&tls.Config{})
+			opts := []grpc.DialOption{grpc.WithTransportCredentials(cred)}
+
+			conn, err := grpc.Dial(fmt.Sprintf("%s:%s", *serverAddr, port), opts...)
+			if err != nil {
+				logger.Printf("Failed to dial for client %v: %v", k, err)
+				return
+			}
+
+			client := pb.NewAllocationServiceClient(conn)
+
+			// Repeat the same experiment for # of runs
+			for r := 0; r < *runs; r++ {
+
+				var wgInner sync.WaitGroup
+
+				for i := 0; i < *perClientCalls; i++ {
+					wgInner.Add(1)
+					go func(i int) {
+						defer wgInner.Done()
+
+						start := time.Now()
+						if err := allocate(client, ts); err != nil {
+
+							muErr.Lock()
+							errorCalls++
+							errorGroup[status.Code(err).String()] = errorGroup[status.Code(err).String()] + 1
+							muErr.Unlock()
```

Comment:
using mutex to add logging and increasing count can impact the performance.
The better way is to create slice/array with the number of clients and let each go routine update their dedicated element un an array